File: sampleCode/CSharp/ConsoleApp/Endpoints/Events/IndustryProprietorIncomeEvent.cs
namespace ConsoleApp.Endpoints.Events;

public record class IndustryProprietorIncomeEvent : Event
{
    public double? Output { get; set; }
    public double? Employment { get; set; }
    public double? EmployeeCompensation { get; set; }
    public required double? ProprietorIncome { get; set; }
    public int IndustryCode { get; set; }

    public override string ImpactEventType => "IndustryProprietorIncome";
}

File: sampleCode/R/Endpoints/Events/IndustryProprietorIncomeEvent.R
source(auth_env$Event)

setClass("IndustryProprietorIncomeEvent",
         contains = "Event",
         slots = list(
           Output = "numeric",
           Employment = "numeric",
           EmployeeCompensation = "numeric",
           ProprietorIncome = "numeric",
           IndustryCode = "integer",
           ImpactEventType = "character"
         ),
         prototype = list(
           ImpactEventType = "IndustryProprietorIncome" # Assigning default values
         ))
